cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+                            )
+                        )
+
+        if self.metric is not None:
+            dims = None
+            if isinstance(self.metric, str):
+                if self.metric in ['diag', 'diag_e']:
+                    self.metric = 'diag_e'
+                elif self.metric in ['dense', 'dense_e']:
+                    self.metric = 'dense_e'
+                else:
+                    if not os.path.exists(self.metric):
+                        raise ValueError('no such file {}'.format(self.metric))
+                    dims = read_metric(self.metric)
+            elif isinstance(self.metric, list):
+                if len(self.metric) != chains:
+                    raise ValueError(
+                        'number of metric files must match number of chains '
+                        ' found {} metric files for {} chains '.format(
+                            len(self.metric), chains
+                        )
+                    )
+                names_set = set(self.metric)
+                if len(names_set) != len(self.metric):
+                    raise ValueError(
+                        'each chain must have its own metric file,'
+                        ' found duplicates in metric files list.'
+                    )
+                for i in range(len(self.metric)):
+                    if not os.path.exists(self.metric[i]):
+                        raise ValueError(
+                            'no such file {}'.format(self.metric[i])
+                        )
+                    if i == 0:
+                        dims = read_metric(self.metric[i])
+                    else:
+                        dims2 = read_metric(self.metric[i])
+                        if len(dims) != len(dims2):
+                            raise ValueError(
+                                'metrics files {}, {},'
+                                ' inconsistent metrics'.format(
+                                    self.metric[0], self.metric[i]
+                                )
+                            )
+                        for j in range(len(dims)):
+                            if dims[j] != dims2[j]:
+                                raise ValueError(
+                                    'metrics files {}, {},'
+                                    ' inconsistent metrics'.format(
+                                        self.metric[0], self.metric[i]
+                                    )
+                                )
+            if dims is not None:
+                if len(dims) > 2 or (len(dims) == 2 and dims[0] != dims[1]):
+                    raise ValueError('bad metric specifiation')
+                self.metric_file = self.metric
+                if len(dims) == 1:
+                    self.metric = 'diag_e'
+                elif len(dims) == 2:
+                    self.metric = 'dense_e'
+
+        if self.adapt_delta is not None:
+            if self.adapt_delta < 0.0 or self.adapt_delta > 1.0:
```

Comment:
.0 are not needed

This is the same:

    if not 0 < self.adapt_delta < 1:

why adapt_delta cannot be between 0 and 1?

Reply:
> why adapt_delta cannot be between 0 and 1?

?? the opposite - it's a percentage - must be between 0 and 1

Reply:
fixed